### PR TITLE
TypedDict completions: tests added

### PR DESCRIPTION
**Pull request by pappasam**
Continues the conversation in this issue: https://github.com/davidhalter/jedi/issues/1478

Note: many tests are failing at present, but not the ones I've added. Notably, one of the tests for actually completing TypedDict keyword parameters that includes quotes is failing weirdly. This tells me there may be an improperly-handled edge case for key completion somewhere. @davidhalter do you have any ideas about where this could be coming from?

```traceback
case = <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:561 "d['fo']">, actual = {"'foo"}, desired = {"'foo'"}

    def assert_case_equal(case, actual, desired):
        """
        Assert ``actual == desired`` with formatted message.

        This is not needed for typical pytest use case, but as we need
        ``--assert=plain`` (see ../pytest.ini) to workaround some issue
        due to pytest magic, let's format the message by hand.
        """
>       assert actual == desired, """
    Test %r failed.
    actual  = %s
    desired = %s
    """ % (case, actual, desired)
E       AssertionError:
E         Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:561 "d['fo']"> failed.
E         actual  = {"'foo"}
E         desired = {"'foo'"}
E
E       assert {"'foo"} == {"'foo'"}
E         Extra items in the left set:
E         "'foo"
E         Extra items in the right set:
E         "'foo'"
E         Use -v to get the full diff
```

### CODE REVIEW

File: test/completion/pep0484_typing.py [hunk 1]
```diff
@@ -509,11 +509,33 @@ def dynamic_annotation(x: int):
...
+    an_int
+
+def typed_dict_test_foo(arg: Foo):
+    a_string = arg['foo']
+    a_list_of_floats = arg['bar']
+    an_int = arg['an_int']
+
+    #? str()
+    a_string
+    #? [float()]
```
Comment (davidhalter) [inline test/completion/pep0484_typing.py]: This line is wrong syntax, you should write `#? float()` instead of `#? [float()]`. For multiple results you could just write `#? float() str()`. It's separated by spaces.

Comment (pappasam) [reply test/completion/pep0484_typing.py]: By `[float()]`, I mean a "list of floats". I don't know how many floats will be present in the result. If `[float()]` is the wrong syntax, how would I express an expected "list of floats"?

**Comment by pappasam** [comment]
Regarding changing the test to `"'foo"` from `"'foo'"`, this sounds simple enough, but I'm curious why the same doesn't need to be done for the test on line 564 with `'"bar"'`? Sounds like there's either a subtle bug we're letting slide, or that there's a meaningful difference between double and single quotes that I don't fully understand.

File: test/completion/pep0484_typing.py (hunk 1 above)
Comment (davidhalter) [reply test/completion/pep0484_typing.py]: It's not really possible to test this AFAIK with just one test. We could obviously make the test algorithm a bit smarter. However I usually test stuff like this with:

```
#? list()
a_list_of_strings
#? float()
a_list_of_strings[undefined_index]
```

BTW: I think you mean `a_list_of_floats`, so the name is wrong :)

**Comment by davidhalter** [comment]
Regarding `foo`. Currently Jedi doesn't add the quote at the end of a string if the next character is a quote. For `'fo'` the next character is `'`. However for `'bar'` it's `r`.

**Comment by pappasam** [comment]
Sadly, looks like integration tests are failing with inheritance. Here are the relevant test failures; hopefully you're able to see where and why things are going weirdly.

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:572 '    foo'> failed.
  actual  = {'test.completion.pep0484_typing:def foo'}
  desired = {'builtins:instance str()'}
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:574 '    an_int'> failed.
  actual  = set()
  desired = {'builtins:instance int()'}

assert set() == {'builtins:instance int()'}
  Extra items in the right set:
  'builtins:instance int()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:583 '    a_string'> failed.
  actual  = set()
  desired = {'builtins:instance str()'}

assert set() == {'builtins:instance str()'}
  Extra items in the right set:
  'builtins:instance str()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:585 '    a_list_of_floats'> failed.
  actual  = set()
  desired = {'builtins:instance list()'}

assert set() == {'builtins:instance list()'}
  Extra items in the right set:
  'builtins:instance list()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:587 '    a_list_of_floats[0]'> failed.
  actual  = set()
  desired = {'builtins:instance float()'}

assert set() == {'builtins:instance float()'}
  Extra items in the right set:
  'builtins:instance float()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:589 '    an_int'> failed.
  actual  = set()
  desired = {'builtins:instance int()'}

assert set() == {'builtins:instance int()'}
  Extra items in the right set:
  'builtins:instance int()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:591 '    another_variable'> failed.
  actual  = set()
  desired = {'builtins:instance int()'}

assert set() == {'builtins:instance int()'}
  Extra items in the right set:
  'builtins:instance int()'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:594 '    a_string.isuppe'> failed.
  actual  = set()
  desired = {'isupper'}

assert set() == {'isupper'}
  Extra items in the right set:
  'isupper'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:596 '    a_list_of_floats.po'> failed.
  actual  = set()
  desired = {'pop'}

assert set() == {'pop'}
  Extra items in the right set:
  'pop'
  Use -v to get the full diff
```

```text
AssertionError:
  Test <IntegrationTestCase: /home/sroeca/src/Personal/jedi/test/completion/pep0484_typing.py:598 '    an_int.as_integer_rati'> failed.
  actual  = set()
  desired = {'as_integer_ratio'}

assert set() == {'as_integer_ratio'}
  Extra items in the right set:
  'as_integer_ratio'
  Use -v to get the full diff
```

File: test/completion/pep0484_typing.py [hunk 2]
```diff
@@ -536,7 +559,41 @@ class Foo(typing.TypedDict):
...
+
+class Bar(Foo):
+    another_variable: int
+
+    #? int()
+    another_variable
+    #? str()
+    foo
+    #? int()
+    an_int
```
Comment (davidhalter) [inline test/completion/pep0484_typing.py]: `foo` and `an_int` should be `#?`. They don't infer anything, this is not a bug, Python would pretty much say the same thing.

Comment (pappasam) [reply test/completion/pep0484_typing.py]: Ah, I see. I've changed the check for `an_int` to `#?` and removed the check for `foo` (it's apparently a function we've defined for other tests, so it has a weird type that's tied to other tests and I think it's prudent to avoid testing for that).

**Comment by davidhalter** [comment]
I'll look into it.

**Comment by davidhalter** [comment]
Merged to master in bb91b96286462299fc222298e36fc7ad6ddd357c

Fixed the issues in 609737322d55090243ffecf0c26c7225859bf83d. The issue was actually that the mro doesn't work anymore if the class is already not a class anymore. I realized that we could just do this when an instance of the TypedDict would be created. For now this works great.

Thanks again for your work on this one!
